### ISSUE TITLE
[Bug]: Chrome - NVDA, I could not navigate between the options using arrow keys (up/down) in Multiselect Dropdown

### ISSUE THREAD

**Issue by binhphan009** [microsoft/fluentui#29089]
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
    Memory: 2.77 GB / 15.73 GB
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (116.0.1938.69)
    Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

Yes

### Reproduction

https://codesandbox.io/s/chrome-nvda-i-could-not-navigate-between-the-options-using-arrow-keys-in-multiselect-dropdown-5ts8dk?file=/index.js

## Actual Behavior
Repro Steps:
1. Open NVDA
2. Focus on to Multi select 
3. click Enter, screen reader will read the first option.
4. Now click on down arrow, to read the second option
Focus does not go to the second option or screen reader does not read the second option
This happens only in NVDA - Chrome
![cannot arrow](https://github.com/microsoft/fluentui/assets/105278576/4978030e-15a3-49d9-a33f-88f1f66fa1a5)

## Expected Behavior
Focus go to the second option or screen reader read the second option

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
